### PR TITLE
initial pass on jaxify pndm scheduler + tests

### PR DESCRIPTION
**Pull request by natolambert**

### CODE REVIEW

**Comment by natolambert** [comment]
FYI Jax precision makes the following test fail by rounding a very small number to zero:

```
  def test_betas(self):
          self.check_over_configs(beta_start=0.01, beta_end=0.2)
```

**Comment by pcuenca** [comment]
I don't think this needs to go into `diffusers@main` yet, as it will require users to install jax to use the `pndm` scheduler. It's cool to have the tests, but then I'd put the implementation in a separate file rather than changing `scheduling_pndm.py`.

File: tests/test_scheduler.py [hunk 1]
```diff
@@ -561,8 +611,9 @@ def test_full_loop_no_noise(self):
             residual = model(sample, t)
             sample = scheduler.step_plms(residual, i, sample)["prev_sample"]
 
-        result_sum = torch.sum(torch.abs(sample))
-        result_mean = torch.mean(torch.abs(sample))
+        import ipdb; pdb.set_trace()
```
Comment (pcuenca) [inline tests/test_scheduler.py:614]: We may need to remove this :)

Comment (natolambert) [reply tests/test_scheduler.py:614]: Oh yeah, I posted the code mostly to share in case anyone wanted to try, but it was a little rude of my to not polish this off.

Comment (pcuenca) [reply tests/test_scheduler.py:614]: No worries!

**Comment by natolambert** [comment]
I agree, this is a proof of concept. Eventually a lot of work would need to be done to make the tests torch, numpy, jax independent. This can be referenced then.
